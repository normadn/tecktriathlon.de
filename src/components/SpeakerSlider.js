import React, { Component } from "react";
import Slider from "react-slick";

import SpeakerCard from "./SpeakerCard";
import SpeakerSliderDots from "./SpeakerSliderDots";

import "slick-carousel/slick/slick.css";
import "slick-carousel/slick/slick-theme.css";

class SpeakerSlider extends Component {
  constructor(props) {
    super(props);
  }

  handleChange = (currentIndex) => this.setState({ currentIndex });

  componentWillMount() {
    this.setState({ currentIndex: 0 });
  }

  onSpeakerSliderDotsClicked = (num) => () => {
    this.slider.slickGoTo(num);
  };

  render() {
    const settings = {
      dots: false,
      arrows: false,
      infinite: false,
      afterChange: (direction) => {
        this.handleChange(direction);
      }
    };

    return (
      <div css={{ width: "100%", position: "relative", zIndex: 3 }}>
        <SpeakerSliderDots
          speakers={this.props.speakers}
          currentIndex={this.state.currentIndex}
          onClick={this.onSpeakerSliderDotsClicked}
        />
        <Slider {...settings} ref={(slider) => (this.slider = slider)}>
          {this.props.speakers.map((speaker) => (
            <SpeakerCard {...speaker} key={speaker.name} />
          ))}
        </Slider>
      </div>
    );
  }
}

export default SpeakerSlider;
