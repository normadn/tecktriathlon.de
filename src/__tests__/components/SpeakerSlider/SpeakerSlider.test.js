import React from "react";
import SpeakerSlider from "../../../components/SpeakerSlider";
import { shallow } from "enzyme";

const speakers = [
  {
    name: "Vinith Suriyakumar",
    title: "Founder and CEO of Clarifai",
    description:
      "Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and ya.",
    image: "coolpic.png"
  },
  {
    name: "Noah Ifergan",
    title: "Founder and CEO of Clarifai",
    description:
      "Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and ya.",
    image: "coolpic.png"
  }
];

describe("SpeakerSlider", () => {
  it("renders correctly", () => {
    expect(shallow(<SpeakerSlider speakers={speakers} />)).toMatchSnapshot();
  });
});
