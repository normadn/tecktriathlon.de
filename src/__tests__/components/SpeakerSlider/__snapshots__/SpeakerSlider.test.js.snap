// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpeakerSlider renders correctly 1`] = `
<div
  css={
    Object {
      "position": "relative",
      "width": "100%",
      "zIndex": 3,
    }
  }
>
  <SpeakerSliderDots
    currentIndex={0}
    onClick={[Function]}
    speakers={
      Array [
        Object {
          "description": "Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and ya.",
          "image": "coolpic.png",
          "name": "Vinith Suriyakumar",
          "title": "Founder and CEO of Clarifai",
        },
        Object {
          "description": "Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and ya.",
          "image": "coolpic.png",
          "name": "Noah Ifergan",
          "title": "Founder and CEO of Clarifai",
        },
      ]
    }
  />
  <Slider
    afterChange={[Function]}
    arrows={false}
    dots={false}
    infinite={false}
  >
    <SpeakerCard
      description="Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and ya."
      image="coolpic.png"
      key="Vinith Suriyakumar"
      name="Vinith Suriyakumar"
      title="Founder and CEO of Clarifai"
    />
    <SpeakerCard
      description="Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and thought leader pioneering the field of applied artificial intelligence (AI). Matt's groundbreaking research in computer vision has propelled the image recognition industry from theory to real-world application. Matthew is a machine learning Ph.D. and ya."
      image="coolpic.png"
      key="Noah Ifergan"
      name="Noah Ifergan"
      title="Founder and CEO of Clarifai"
    />
  </Slider>
</div>
`;
